### PR TITLE
[BUGFIX beta] Update VM, fix component name preprocessing

### PR DESCRIPTION
**Pull request by pzuraq**

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/environment.ts [hunk 1]
```diff
@@ -50,29 +51,36 @@ setGlobalContext({
...
-      deprecate(message, false, {
-        id: 'autotracking.mutation-after-consumption',
-        until: '4.0.0',
-        for: 'ember-source',
-        since: {
-          enabled: '3.21.0',
-        },
-      });
-    },
+      let override = VM_ASSERTION_OVERRIDES.find((o) => o.id === id);
```
Comment (rwjblue) [inline packages/@ember/-internals/glimmer/lib/environment.ts:59]: Can't use `.find` on IE11 (this is failing browser stack)

Comment (rwjblue) [reply packages/@ember/-internals/glimmer/lib/environment.ts:59]: I think we should probably change the overrides to be a map / pojo where the key is the `id`.  What do you think?

File: packages/@ember/-internals/glimmer/lib/environment.ts [hunk 2]
```diff
@@ -50,29 +51,36 @@ setGlobalContext({
...
+
+      assert(override?.message ?? msg, test);
+    }
+  },
+
+  deprecate(msg: string, test: unknown, options: { id: string }) {
+    if (DEBUG) {
+      let { id } = options;
+
+      let override = VM_DEPRECATION_OVERRIDES.find((o) => o.id === id);
```
Comment (rwjblue) [inline packages/@ember/-internals/glimmer/lib/environment.ts:69]: Same here RE: `.find`

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Comment by jamescdavis** [comment]
Fixes #19360

**Review by rwjblue — APPROVED** [review]

**Pull request merged**